storage_service/locations/models/wellcome.py

```diff
@@ -69,14 +69,42 @@ def browse(self, path):
             'properties': {},
         }
 
+    @property
+    def wellcome_client(self):
+        return StorageServiceClient(
+            api_url=self.api_root_url,
+            token_url=self.token_url,
+            client_id=self.app_client_id,
+            client_secret=self.app_client_secret,
+        )
+
     def delete_path(self, delete_path):
         LOGGER.debug('Deleting %s from Wellcome storage' % delete_path)
 
     def move_to_storage_service(self, src_path, dest_path, dest_space):
         """ Moves src_path to dest_space.staging_path/dest_path. """
-        LOGGER.debug('Fetching %s from %s (%s) on Wellcome storage' % (
+        LOGGER.debug('Fetching %s on Wellcome storage to %s (space %s)' % (
             src_path, dest_path, dest_space))
 
+        space_id, source_id = src_path.lstrip('/').split('/')
+
+        bag = self.wellcome_client.get_bag(space_id, source_id)
+        for loc in bag['locations']:
+            if loc['provider']['id'] == 'aws-s3-ia':
+                LOGGER.debug("Fetching files from s3://%s%s", loc['bucket'], loc['path'])
+                bucket = self.s3_resource.Bucket(loc['bucket'])
+
+                # The bag is stored unzipped (i.e. as a directory tree).
+                # Download all objects in the source directory
+                s3_prefix = loc['path'].lstrip('/')
+                objects = bucket.objects.filter(Prefix=s3_prefix)
+                for objectSummary in objects:
+                    dest_file = objectSummary.key.replace(s3_prefix, dest_path, 1)
+                    self.space.create_local_directory(dest_file)
+
+                    bucket.download_file(objectSummary.key, dest_file)
+                break
```

Comment:
Maybe worth throwing an error here if we don't provide you with an `aws-s3-ia` location?